### PR TITLE
feature/8081-rachael-detox-automatically-run-on-PR

### PR DESCRIPTION
**Pull request by rbontrager**
## Description of Change

Add the ability to run the relevant detox test automatically on a PR

## Testing

- [ ] Tested on iOS 
- [ ] Tested on Android 

## PR Checklist

  **Reviewer:** Confirm the items below as you review

- [ ] PR is connected to issue(s)
- [ ] Tests are included to cover this change (when possible)
- [ ] No magic strings (All string unions follow the [Union -> Constant](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/develop/VAMobile/src/constants/common.ts) type pattern)
- [ ] No secrets or API keys are checked in
- [ ] All imports are absolute (no relative imports)
- [ ] New functions and Redux work have proper TSDoc annotations

## For QA

[Run a build for this branch](https://github.com/department-of-veterans-affairs/va-mobile-app/actions/workflows/on_demand_build.yml)

### CODE REVIEW

File: .github/workflows/e2e_android.yml [hunk 1]
```diff
@@ -7,7 +7,7 @@ name: '[e2e] Android Detox Test Run'
 on:
   push:
     branches:
-      - 'dependabot/**'
+      - develop
```
Comment (Sparowhawk) [inline .github/workflows/e2e_android.yml:10]: is the objective to run the detox tests after it is pushed to develop? That is what this line would do

File: .github/workflows/e2e_ios.yml [hunk 2]
```diff
@@ -7,7 +7,7 @@ name: '[e2e] iOS Detox Test Run'
 on:
   push:
     branches:
-      - 'dependabot/**'
+      - develop
```
Comment (Sparowhawk) [inline .github/workflows/e2e_ios.yml:10]: is the objective to run the detox tests after it is pushed to develop? That is what this line would do

File: .github/workflows/e2e_android.yml (hunk 1 above)
Comment (Sparowhawk) [reply .github/workflows/e2e_android.yml:10]: if that is the objective here is to run after a pr is merged then I think this is good to go

Comment (rbontrager) [reply .github/workflows/e2e_android.yml]: The objective is to have it run whenever someone makes a change to their PR.  I know we talked about it (and I forgot to take the code out) so I think its correct now but correct me if I'm still wrong.

**Review by Sparowhawk — APPROVED** [review]

**Comment by lindluni** [comment]
Hello all :wave: 

@lindluni here with the VA GitHub Admin team.

This PR results in your `e2e_android` tests consuming approximately $1,500/day in consumption to run just this one workflow, which is more that we spend on Actions combined across all of our other repositories. A lot of this has to do with not cancelling existing workflows when pushing new commits to existing PR's, thus unnecessarily running duplicate workflows. 

To do this, you would need to add the `concurrency` group key and properly split your workflow to cancel your existing PR runs when pushing new commits.

We'd ask that you revert this change and open an issue in the [https://github.com/department-of-veterans-affairs/github-user-requests](https://github.com/department-of-veterans-affairs/github-user-requests) repository so we can help you add the proper concurrency group to your PR workflow to protect our Actions spend.

Thank you for your help with this.

**Pull request merged**